Dimensions are not filled for movies

patrickbroens opened the pull request:
For calculating widths and heights in a gallery, like in Text and Media, the dimensions of movies are necessary. YouTube does not provide this data in gdata, as used in the YouTubeHelper, but does provide dimensions in oembed. Vimeo returns it out of the box.

patrickbroens committed:
[BUGFIX] Dimensions are not filled for YouTube movies

For calculating widths and heights in a gallery, like in Text and Media, the dimensions of movies are necessary. YouTube does not provide this data in gdata, as used in the YouTubeHelper, but does provide dimensions in oembed.

This patch adds a call to oembed next to the existing gdata call, for the only purpose to get the dimensions. The width and height meta data for the YouTube movie is filled with these dimensions.

patrickbroens committed:
[BUGFIX] Dimensions are not filled for Vimeo movies

For calculating widths and heights in a gallery, like in Text and Media, the dimensions of movies are necessary. Vimeo does provide the dimensions out of the box.

With this patch the width and height metadata for a Vimeo movie are filled

patrickbroens committed:
[TASK] Checking if width and height metadata exist is not needed

When setting the width and height for a YouTube movie, the existing meta data is checked if these values are filled. This is not needed.

This patch removes these checks

liayn commented:
Why not using the V3 API in general, which provides all information?
also consider https://developers.google.com/api-client-library/php/

patrickbroens commented:
When using the V3 API, you need an API key and the requests are limited. Most of the information must be made public by the author, otherwise you get hardly any information. The exact upload dimensions are not retrievable anymore, only the aspect ratio, which is actually not right. When you upload a video with non-standard dimensions, YouTube is just guessing the aspect ratio. As long as Google keeps up the old API's, I would suggest to stick to it. Especially for users, because otherwise they need to apply for a developer API key. I hope they will add more features to V3 when it becomes stable.

patrickbroens committed:
[TASK] Use https for Google YouTube Oembed

The YouTube helper is calling gdata with https and oembed with http

This patch changes the call to oembed to https as well

patrickbroens commented:
oembed call has been changed to https

liayn commented:
You're right about the key of course. Keep it like that! Sorry for not thinking about that

fsaris commented:
tnx for the patch :)
